src/blocks/scratch3_looks.js

```diff
@@ -344,73 +344,94 @@ class Scratch3LooksBlocks {
     }
 
     /**
-     * Utility function to set the costume or backdrop of a target.
+     * Utility function to set the costume of a target.
      * Matches the behavior of Scratch 2.0 for different types of arguments.
-     * @param {!Target} target Target to set costume/backdrop to.
+     * @param {!Target} target Target to set costume to.
      * @param {Any} requestedCostume Costume requested, e.g., 0, 'name', etc.
      * @param {boolean=} optZeroIndex Set to zero-index the requestedCostume.
      * @return {Array.<!Thread>} Any threads started by this switch.
      */
-    _setCostumeOrBackdrop (target,
-        requestedCostume, optZeroIndex) {
+    _setCostume (target, requestedCostume, optZeroIndex) {
         if (typeof requestedCostume === 'number') {
-            target.setCostume(optZeroIndex ?
-                requestedCostume : requestedCostume - 1);
+            target.setCostume(optZeroIndex ? requestedCostume : requestedCostume - 1);
         } else {
-            const costumeIndex = target.getCostumeIndexByName(requestedCostume);
-            if (costumeIndex > -1) {
+            const costumeIndex = target.getCostumeIndexByName(requestedCostume.toString());
+
+            if (costumeIndex !== -1) {
                 target.setCostume(costumeIndex);
-            } else if (requestedCostume === 'previous costume' ||
-                       requestedCostume === 'previous backdrop') {
-                target.setCostume(target.currentCostume - 1);
-            } else if (requestedCostume === 'next costume' ||
-                       requestedCostume === 'next backdrop') {
+            } else if (requestedCostume === 'next costume') {
                 target.setCostume(target.currentCostume + 1);
-            } else if (requestedCostume === 'random backdrop') {
-                const numCostumes = target.getCostumes().length;
+            } else if (requestedCostume === 'previous costume') {
+                target.setCostume(target.currentCostume - 1);
+            } else if (!isNaN(requestedCostume) &&
+                        (typeof requestedCostume !== 'string' || /\d/g.test(requestedCostume))) {
```

Comment:
I'm not sure why you need the regex test here.

If you're tying to see if it's a string representation of a number e.g. "12", you could try casting requestedCostume to a number and see if it results in `NaN` or not.

Reply:
e.g. in the [old code here](https://github.com/LLK/scratch-vm/pull/1517/files?utf8=%E2%9C%93&diff=split#diff-8d62e377c35d800fc8cfe63d4e98232fL376)

Reply:
Hmmm it seems like maybe we could use a combination of `parseInt` and `isNaN`. Though, unfortunately `parseInt` does not work the way we want for boolean inputs, so we would have to deal with those explicitly.

I'm also okay with leaving this code as is now that I know the reason.

In either case, could you add a comment here explaining the digit check (e.g. the fact that whitespace is treated as 0 when casting to a number, and we don't want that)

Reply:
This is a good point. I think I'll use the `Cast` utility for this -- `Cast.isWhiteSpace`.

Reply:
Since I'm now using the very readable function, I decided to remove the explanation that JS casts whitespace to numbers.